pkg/event/processor_test.go

```diff
@@ -449,3 +449,39 @@ func TestChanQueueEventProcessor_ProcessBatch(t *testing.T) {
 		assert.True(t, len(logEvent.Event.Visitors) >= 1)
 	}
 }
+
+func BenchmarkProcessor(b *testing.B) {
+	exeCtx := utils.NewCancelableExecutionCtx()
+	processor := NewBatchEventProcessor(
+		WithQueueSize(1000),
```

Comment:
Following the example in [this](https://dave.cheney.net/2013/06/30/how-to-write-benchmarks-in-go) article you can parameterize the benchmark.
```go
func benchmarkProcessor(qs int, b *testing.B) {
	exeCtx := utils.NewCancelableExecutionCtx()
	processor := NewBatchEventProcessor(
		WithQueueSize(qs)
		WithFlushInterval(10 * time.Millisecond),
		WithEventDispatcher(NewMockDispatcher(100, false)))
	processor.Start(exeCtx)

	impression := BuildTestImpressionEvent()
	conversion := BuildTestConversionEvent()

	for i := 0; i < b.N; i++ {
		processor.ProcessEvent(impression)
		processor.ProcessEvent(conversion)
		time.Sleep(500)
	}
	processor.Stop()
}

func BenchmarkQueueSize100(b *testing.B) { benchmarkProcessor(100, b) }
func BenchmarkQueueSize2000(b *testing.B) { benchmarkProcessor(2000, b) }
func BenchmarkQueueSize5000(b *testing.B) { benchmarkProcessor(5000, b) }
func BenchmarkQueueSize10000(b *testing.B) { benchmarkProcessor(10000, b) }
func BenchmarkQueueSize50000(b *testing.B) { benchmarkProcessor(50000, b) }
```